pkg/files/files.go

```diff
@@ -0,0 +1,92 @@
+package files
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"mime/multipart"
+	"os"
+
+	appliancepkg "github.com/appgate/sdpctl/pkg/appliance"
+	"github.com/appgate/sdpctl/pkg/tui"
+	"github.com/cenkalti/backoff/v4"
+	log "github.com/sirupsen/logrus"
+)
+
+type FilesAPI struct {
+	API      *appliancepkg.Appliance
+	Progress *tui.Progress
+}
+
+func (f *FilesAPI) Upload(ctx context.Context, file *os.File) error {
+	fileInfo, err := file.Stat()
+	if err != nil {
+		return err
+	}
+	size := fileInfo.Size()
+	name := fileInfo.Name()
```

Comment:
I think it would be good to allow, optional, overriding the file name used when uploading.

So you can potentially have one file name locally and it ends up being uploaded with a different name on the controller file repository.

I have a feeling that could be useful in some support cases.

Reply:
Ok, it's added now. The UX is not obvious, because you can upload multiple files at once, but it works pretty well and is documented in the help.